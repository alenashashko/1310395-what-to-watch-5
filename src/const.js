export const MillisecondsInTimePeriod = {
  SECOND: 1000,
  MINUTE: 60000,
  HOUR: 3600000,
};
