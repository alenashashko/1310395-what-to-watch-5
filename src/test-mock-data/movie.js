export const movie = {
  id: `1`,
  title: `Orlando`,
  picture: `/img/orlando.jpg`,
  backgroundPicture: `/img/orlando.jpg`,
  backgroundColor: `#e1b0b2`,
  poster: `/img/orlando.jpg`,
  genre: `Crime`,
  year: 2015,
  ratingScore: 7.8,
  ratingCount: 1000,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 110,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
  Sed sed nisi sed augue convallis suscipit in sed felis.
  Aliquam erat volutpat.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`,
    `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  isFavorite: false
};

export const movies = [{
  id: `1`,
  title: `Orlando`,
  picture: `/img/orlando.jpg`,
  backgroundPicture: `/img/orlando.jpg`,
  backgroundColor: `#e1b0b2`,
  poster: `/img/orlando.jpg`,
  genre: `Crime`,
  year: 2015,
  ratingScore: 7.8,
  ratingCount: 1000,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 110,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
  Sed sed nisi sed augue convallis suscipit in sed felis.
  Aliquam erat volutpat.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`,
    `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  isFavorite: false
},
{
  id: `2`,
  title: `Revenant`,
  picture: `/img/revenant.jpg`,
  backgroundPicture: `/img/revenant.jpg`,
  backgroundColor: `#bdad8f`,
  poster: `/img/revenant.jpg`,
  genre: `Romance`,
  year: 2010,
  ratingScore: 5.9,
  ratingCount: 850,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 145,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
  director: `Robert Zemeckis`,
  starring: [`Will Smith`, `Michael J. Fox`, `Harrison Ford`,
    `Sandra Bullock`, `Betty White`, `Tom Hanks`, `Denzel Washington`],
  isFavorite: false
}
];

export const genres = [`Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`,
  `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`];

export const genre = `All genres`;
