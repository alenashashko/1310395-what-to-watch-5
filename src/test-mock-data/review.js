import {adaptCommentToClient} from '../services/adapters';

export const apiReviews = [
  {
    id: 1,
    user: {
      id: 11,
      name: `Jack`
    },
    rating: 2.4,
    comment: `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    date: `2020-11-03T13:38:44.769Z`
  },
  {
    id: 2,
    user: {
      id: 12,
      name: `Isaac`
    },
    rating: 7.2,
    comment: `A movie that will take you to another world full of emotions.`,
    date: `2020-10-23T13:38:44.769Z`
  }
];

export const review = adaptCommentToClient(apiReviews[0]);

export const reviews = apiReviews.map(adaptCommentToClient);

