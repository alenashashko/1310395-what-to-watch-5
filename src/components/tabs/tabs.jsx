import React from 'react';
import PropTypes from 'prop-types';

import proptypes from '../../type';
import MovieOverview from '../movie-overview/movie-overview';
import MovieDetails from '../movie-details/movie-details';
import MovieReviews from '../movie-reviews/movie-reviews';
import {TabTypes, tabs} from '../../const';
import withActiveTab from '../../hocs/with-active-tab/with-active-tab';

const Tabs = (props) => {
  const {activeTab, onTabClick, movie, reviews} = props;
  // id фильма

  const getComponentByTab = (tab) => {
    switch (tab) {
      case TabTypes.DETAILS:
        return <MovieDetails movie={movie} />;
      case TabTypes.REVIEWS:
        return <MovieReviews reviews={reviews} />;
      default:
        return <MovieOverview movie={movie} />;
    }
  };

  return (
    <React.Fragment>
      <nav className="movie-nav movie-card__nav">
        <ul className="movie-nav__list">
          {tabs.map((tab) =>
            <li
              key={tab}
              className={`movie-nav__item ${activeTab === tab ? `movie-nav__item--active` : ``}`}>
              <a href="#" className="movie-nav__link"
                onClick={(evt) => {
                  evt.preventDefault();
                  onTabClick(tab);
                }}>
                {tab}
              </a>
            </li>)}
        </ul>
      </nav>
      {getComponentByTab(activeTab)}
    </React.Fragment>
  );
};

Tabs.propTypes = {
  activeTab: PropTypes.string.isRequired,
  onTabClick: PropTypes.func.isRequired,
  movie: proptypes.movie,
  reviews: proptypes.reviews
};

export default withActiveTab(Tabs);

