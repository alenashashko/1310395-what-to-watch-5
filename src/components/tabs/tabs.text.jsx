import React from 'react';
import renderer from 'react-test-renderer';

import {Tabs} from './tabs';
import {movie} from '../../test-mock-data/movie';
import {TabType} from '../../const';

const noop = () => {};

describe(`Should Tabs render correctly`, () => {
  it(`with DETAILS active tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            activeTab={TabType.DETAILS}
            onTabClick={noop}
            movie={movie}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`with OVERVIEW active tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            activeTab={TabType.OVERVIEW}
            onTabClick={noop}
            movie={movie}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`with REVIEWS active tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            activeTab={TabType.REVIEWS}
            onTabClick={noop}
            movie={movie}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
