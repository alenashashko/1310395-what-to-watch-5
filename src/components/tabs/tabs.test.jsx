import React from 'react';
import renderer from 'react-test-renderer';

import {Tabs} from './tabs';
import {movie} from '../../test-mock-data/movie';
import {TabType} from '../../const';

const noop = () => {};

jest.mock(`../movie-reviews/movie-reviews`, () => {
  const MovieReviewsMock = () => <p>MovieReviewsMock</p>;

  return {
    __esModule: true,
    default: MovieReviewsMock
  };
});

describe(`Should Tabs render correctly`, () => {
  it(`with DETAILS active tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            activeTab={TabType.DETAILS}
            onTabClick={noop}
            movie={movie}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`with OVERVIEW active tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            activeTab={TabType.OVERVIEW}
            onTabClick={noop}
            movie={movie}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`with REVIEWS active tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            activeTab={TabType.REVIEWS}
            onTabClick={noop}
            movie={movie}
          />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
