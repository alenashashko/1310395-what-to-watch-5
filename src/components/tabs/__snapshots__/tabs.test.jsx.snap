// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Tabs render correctly with DETAILS active tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Nicolas Winding Refn
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Tom Hardy
          ,
          <br />
          Kelly Adams
          ,
          <br />
          Luing Andrews
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          1h 32m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Action
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2008
        </span>
      </p>
    </div>
  </div>,
]
`;

exports[`Should Tabs render correctly with OVERVIEW active tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      3.6
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Normal
      </span>
      <span
        className="movie-rating__count"
      >
        109661
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Nicolas Winding Refn
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tom Hardy, Kelly Adams, Luing Andrews
      </strong>
    </p>
  </div>,
]
`;

exports[`Should Tabs render correctly with REVIEWS active tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <p>
    MovieReviewsMock
  </p>,
]
`;
